crates/pathfinder/src/cairo/ext_py.rs

```diff
@@ -74,6 +114,31 @@ pub enum CallFailure {
     Shutdown,
 }
 
+/// Where should the call code get the used `BlockInfo::gas_price`
+#[derive(Debug)]
+pub enum GasPriceSource {
+    /// Use gasPrice recorded on the `starknet_blocks::gas_price`.
+    ///
+    /// This is not implied by other arguments such as `at_block` because we might need to
+    /// manufacture a block hash for some future use cases.
+    PastBlock,
+    /// Use this latest value from `eth_gasPrice`.
+    ///
+    /// U256 is not used for serialization matters, [u8; 32] could be used as well. python side's
+    /// serialization limits this value to u128 but in general `eth_gasPrice` is U256.
+    Current(web3::types::H256),
```

Comment:
> 2^128 - 1
The only place I saw that was with the long field metadata, and I think an explicit (de)serialization point is required to run that field metadata validation, otherwise, no one reads the fields.